multi-agent/lib/mas/prompts/ports.py

```
@@ -0,0 +1,82 @@
+    def describe_batch(self, schedule_ids: List[str]) -> Dict[str, Optional[ScheduleInfo]]:
+        """Batch read-back of multiple schedules' live state.
+
+        Returns a mapping from schedule_id to ScheduleInfo (or None if not found).
+        Default implementation falls back to sequential describe() calls.
+        """
+        results: Dict[str, Optional[ScheduleInfo]] = {}
+        for sid in schedule_ids:
+            try:
+                results[sid] = self.describe(sid)
+            except ScheduleNotFoundError:
+                results[sid] = None
```

Ruff (0.15.21)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)